src/end.sol

```diff
@@ -283,11 +312,29 @@ contract End is LibNote {
         tag[ilk] = wdiv(spot.par(), uint(pip.read()));
     }
 
+    function halt(bytes32 ilk, uint256 id) external note {
+        require(tag[ilk] != 0, "End/tag-ilk-not-defined");
+
+        (address _clip,,,) = dog.ilks(ilk);
+        ClipLike clip = ClipLike(_clip);
+        (, uint rate,,,) = vat.ilks(ilk);
+        (, uint256 tab, uint256 lot, address usr,,) = clip.sales(id);
+        
+        vat.suck(address(vow), address(vow),  tab);
+        vat.hope(address(clip)); // Should we have this?
```

Comment:
I don't see why this is needed. It's necessary in `skip` since there's a `move` call that transfer `dai` away from the caller, but `Clipper.yank` just `fluxes` collateral to the caller which doesn't require any special permission.